### ISSUE TITLE
[FIRRTL] Context Dependent Port Parsing

### ISSUE THREAD

**Issue by seldridge** [llvm/circt#559]
The following circuit is currently valid, but fails parsing:

```scala
circuit Tricky:
  module Tricky:
    input input: UInt<1>
    output output: UInt<1>
    output <= input
```

```bash
firtool Tricky.fir
# Tricky.fir:5:12: error: expected port name
#     output <= input
```

**Comment by lattner** [comment]
Fixed in fa5aade, thanks!

**Comment by seldridge** [comment]
Awesome. Thanks for knocking this out, @lattner!
